File: app/src/main/java/com/github2136/mvp/presenter/MainPresenter.java
package com.github2136.mvp.presenter;

import com.github2136.base.BaseMVPPresenter;
import com.github2136.mvp.model.UserModel;
import com.github2136.mvp.ui.activity.MainActivity;

import okhttp3.Response;

/**
 *
 **/
public class MainPresenter extends BaseMVPPresenter<MainActivity> {
    private UserModel mUserModel;

    public MainPresenter(MainActivity mainActivity) {
        super(mainActivity);
        mUserModel = new UserModel(mainActivity.getApplicationContext(), mainActivity.toString());
    }

    public void get() {
        mUserModel.get(new HttpCallback() {

            @Override
            protected void onFailure(MainActivity mainActivity, String str) {
                mainActivity.getFailure(str);
            }

            @Override
            protected void onResponse(MainActivity mainActivity, Response response, String bodyStr) {
                if (response.isSuccessful()) {
                    mainActivity.getSuccessful(bodyStr);
                } else {
                    mainActivity.getFailure(failedStr);
                }
            }
        });
    }

    @Override
    public void cancelRequest() {
        mUserModel.cancelRequest();
    }
}

File: app/src/main/java/com/github2136/mvp/presenter/MainPresenter.kt
package com.github2136.mvp.presenter

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.github2136.base.BaseMVPPresenter
import com.github2136.base.CallbackData
import com.github2136.mvp.model.DataModel

/**
 * Created by yb on 2018/11/2.
 */
class MainPresenter(private val app: Application) : BaseMVPPresenter(app) {
    private val mDataModel: DataModel  by lazy { DataModel(app, mTag) }
    val ldGet: MutableLiveData<CallbackData> = MutableLiveData()

    fun get() {
        mDataModel.getStr(
                response = { _, response ->
                    if (response.isSuccessful) {
                        ldGet.postValue(CallbackData(true, response.body()?.string()))
                    } else {
                        ldGet.postValue(CallbackData(false, failedStr))
                    }
                },
                failure = { _, _ -> ldGet.postValue(CallbackData(false, failedStr)) }
        )
    }

    override fun cancelRequest() {
        mDataModel.cancelRequest()
    }
}